### DISCUSSION TITLE
How can I serialize and deserialize List<Dictionary<string, string>>within AOT?

### DISCUSSION THREAD

**Discussion by BoyFaceGirl** [dotnet/runtime#107072] [Q&A]
Do you have code examples?

**Comment by Joe4evr** [comment]
For what format?

If it's JSON, then use `System.Text.Json` and [set up your code to use the source generation feature](https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/source-generation?pivots=dotnet-8-0).
It should look something like this:
```cs
[JsonSerializable(typeof(List<Dictionary<string, string>>))]
public partial class YourSerializerContext : JsonSerializerContext { }
```
Then you can pass `YourSerializerContext.Default` to the `Serialize`/`Deserialize` methods.

**Reply by BoyFaceGirl** [reply]
> For what format?
> 
> If it's JSON, then use `System.Text.Json` and [set up your code to use the source generation feature](https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/source-generation?pivots=dotnet-8-0). It should look something like this:
> 
> ```cs
> [JsonSerializable(typeof(List<Dictionary<string, string>>))]
> public partial class YourSerializerContext : JsonSerializerContext { }
> ```
> 
> Then you can pass `YourSerializerContext.Default` to the `Serialize`/`Deserialize` methods.

If the value of the dictionary is' object ', the datetime type serialization will fail. What should I do?
